### PR TITLE
ANC: swift concurrency

### PR DESCRIPTION
**Pull request by tonikocjan**
Concurrency (async/await) support for ANC. https://github.com/poviolabs/PovioKit/issues/218

Example:

```swift
struct ChuckNorris: Decodable {
  let iconUrl: URL
  let value: String
}

let client = AlamofireNetworkClient()

do {
  let chuck = try await client
    .request(method: .get, endpoint: "https://api.chucknorris.io/jokes/random")
    .decodeAsync(ChuckNorris.self, decoder: decoder)
  print(chuck)
} catch {
  print(error)
}

do {
  async let chuck1 = client
    .request(method: .get, endpoint: "https://api.chucknorris.io/jokes/random")
    .decodeAsync(ChuckNorris.self, decoder: decoder)
  async let chuck2 = client
    .request(method: .get, endpoint: "https://api.chucknorris.io/jokes/random")
    .decodeAsync(ChuckNorris.self, decoder: decoder)
  async let chuck3 = client
    .request(method: .get, endpoint: "https://api.chucknorris.io/jokes/random")
    .decodeAsync(ChuckNorris.self, decoder: decoder)

  let allChucks = try await [chuck1, chuck2, chuck3]
  for chuck in allChucks {
    print(chuck)
  }
} catch {
  print(error)
}
```

### CODE REVIEW

File: Sources/PromiseKit/Promise.swift [hunk 1]
```diff
@@ -865,6 +865,24 @@ public extension Promise where Value: OptionalType {
   }
 }
 
+public extension Promise {
+  /// Convert promise to async/await
+  var asAsync: Value {
```
Comment (borut-t) [inline Sources/PromiseKit/Promise.swift:870]: Neat 👌

**Review by borut-t — APPROVED** [review]
Looks great and simple. We should try it on the project.

**Review by markom-povio — APPROVED** [review]

**Comment by borut-t** [comment]
@tonikocjan, I think this should work: 
```swift
func decode<D: Decodable>(
  _ decodable: D.Type,
  decoder: JSONDecoder = .init(),
  on dispatchQueue: DispatchQueue? = .main
) -> Promise<D> {
  ...
}

func decode<D: Decodable>(
  _ decodable: D.Type,
  decoder: JSONDecoder = .init(),
  on dispatchQueue: DispatchQueue? = .main
) async throws -> D {
  ...
}
```
This way we could still differentiate between the two methods...
```swift
// promise based pattern
.decode().finally { ... }

or

// async/await
Task {
  try await decode()
}
```

File: Sources/Networking/AlamofireNetworkClient/AlamofireNetworkClient.swift [hunk 2]
```diff
@@ -354,6 +370,108 @@ public extension AlamofireNetworkClient.Request {
...
+  var asStringAsync: String {
+    get async throws {
+      try await string()
+    }
+  }
+  
+  /// See Alamofire's documentation for details:
+  /// 
+  /// ``Alamofire.Concurrency.serializingDecodable``
+  func decod<D: Decodable>(
```
Comment (borut-t) [inline Sources/Networking/AlamofireNetworkClient/AlamofireNetworkClient.swift:448]: Typo `decod`

**Review by borut-t — APPROVED** [review]
👌

**Pull request merged**